sklearn/utils/_array_api.py

```diff
@@ -669,6 +669,40 @@ def _average(a, axis=None, weights=None, normalize=True, xp=None):
     return sum_ / scale
 
 
+def _median(x, axis=None, keepdims=False, xp=None):
+    # `median` is not included in the Array API spec, but is implemented in most
+    # array libraries (and all that we test).
+    xp, _ = get_namespace(x, xp=xp)
+    if hasattr(xp, "median"):
+        # When `x` has even number of elements, `torch.median` takes the lower of the
+        # two medians, thus we use `torch.quantile(q=0.5)`, which gives mean of the two
+        if _is_xp_namespace(xp, "torch"):
+            return xp.quantile(x, q=0.5, dim=axis, keepdim=keepdims)
+        return xp.median(x, axis=axis, keepdims=keepdims)
+
+    if x.ndim == 0:
```

Comment:
I was unsure of what to do here. All array libraries 'actively considered' (https://data-apis.org/array-api/latest/purpose_and_scope.html#stakeholders) have a `median` implementation, so all arrays that we test have a `median` implementation.

This means that this will really never be tested or used, and I am tempted to just remove?

Reply:
Nevermind, we do need the implementation for array-api-strict :facepalm:

Reply:
If this is just for array-api-strict I think we can convert the arrays to numpy and use numpy median and then convert the result back to the correct array type before returning. Implementing this seems a bit extra as I doubt array-api-strict would be used in practice to speed up computations.

Reply:
Yes you're right, don't know why I didn't think of that :facepalm: